docs/python/debugging.md

```diff
@@ -55,11 +55,214 @@ Selecting the **Node.js: Gulp task** yields the following result:
+
+1. Verify that you can see a prompt in the SSH session.
+
+1. In your VS Code workspace, create a configuration for remote debugging in your `launch.json` file, setting the port to match the port used in the `ssh` command and the host to `localhost`. You use `localhost` here because you've set up the SSH tunnel.
+
+```json
+{
+    "name": "Python: Attach",
+    "type": "python",
+    "request": "attach",
+    "port": 5678,
+    "host": "localhost",
+    "pathMappings": [
+        {
+            "localRoot": "${workspaceFolder}", // Maps C:\Users\user1\project1
+            "remoteRoot": "."                  // To current working directory ~/project1
+        }
+    ]
+}
+```
+
+**Starting debugging**
+
+Now that an SSH tunnel has been set up to the remote computer, you can begin your debugging.
+
+1. Both computers: make sure that identical source code is available.
+
+1. Both computers: install [debugpy](https://pypi.org/project/debugpy/) using `python -m pip install --upgrade debugpy` into your environment (while using a form of virtual environment is not required, it is a recommended best practice).
+
+1. Remote computer: there are two ways to specify how to attach to the remote process.
+
+   1. In the source code, add the following lines, replacing `address` with the remote computer's IP address and port number (IP address 1.2.3.4 is shown here for illustration only).
+
+        ```python
+        import debugpy
+
+        # Allow other computers to attach to debugpy at this IP address and port.
+        debugpy.listen(('1.2.3.4', 5678))
+
+        # Pause the program until a remote debugger is attached
+        debugpy.wait_for_client()
+        ```
+
+        The IP address used in `listen` should be the remote computer's private IP address. You can then launch the program normally, causing it to pause until the debugger attaches.
+
+   1. Launch the remote process through debugpy, for example:
+
+       ```bash
+       python3 -m debugpy --listen 5678 --wait-for-client -m myproject
+       ```
+
+       This starts the package `myproject` using `python3`, with the remote computer's private IP address of `1.2.3.4` and listening on port `5678` (you can also start the remote Python process by specifying a file path instead of using `-m`, such as `./hello.py`).
+
+1. Local computer: **Only if you modified the source code on the remote computer as outlined above**, then in the source code, add a commented-out copy of the same code added on the remote computer. Adding these lines makes sure that the source code on both computers matches line by line.
```

Comment:
I think the more usual workflow is to modify the code locally first, and then deploy it to the remote machine with those modifications. Having mismatched code between local and remote, even when line numbers match, is generally not a good idea, and we might be more stringent about it in the future (e.g. do checksums).

Reply:
That makes sense as a better flow. I was just trying to maintain the content that was already there. I'll flag this as an additional improvement to make once we get everything else updated.